src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -181,18 +181,33 @@ def _process_resource_change(
     # The class is serialized as a string when enqueueing the class.
     model: type[Event] | type[Model] = TYPES[sender]
     instance: Event | Model | None = None
-    # The Event model has different hooks for the different event types. The sender
-    # determines which type eg. Error and therefore the 'name' eg. error
-    if issubclass(model, Event):
-        if not kwargs.get("instance"):
-            extra = {"sender": sender, "action": action, "event_id": instance_id}
-            logger.info("process_resource_change.event_missing_event", extra=extra)
-            return
+
+    project_id: int | None = kwargs.get("project_id", None)
+    group_id: int | None = kwargs.get("group_id", None)
+    if sender == "Error" and not kwargs.get("instance", None) and project_id and group_id:
+        # Read event from nodestore as we're trying to move away from passing events in task
+        # messages.
+        node_id = Event.generate_node_id(project_id, str(instance_id))
+        nodedata = nodestore.backend.get(node_id)
+        node_event = Event(
```

Comment:
Is there a way to distinguish GroupEvents vs Events ? It seems like we can also get a GroupEvent for this task ( though 182 & 203/218 🫠 )

Reply:
You'll always be fine to just fetch an `Event` here, since we only ever pass error events.

Reply:
I see two calls for `process_resource_change_bound`. Both are in post_process_group https://github.com/getsentry/sentry/blob/300cef1ecc434aceddfabdd02ba1b0f5aff6ea17/src/sentry/tasks/post_process.py#L1166-L1173

Where does `GroupEvent` get in?

Reply:
Post process group actually converts the `Event` into a `GroupEvent` for consistency. 

https://github.com/getsentry/sentry/blob/cdbc6c38a6b8244c01accd87e1e314fac4139438/src/sentry/tasks/post_process.py#L623

The main difference is that a `GroupEvent` can contain an occurrence. Since these are only error events, then the `Event` and `GroupEvent` are basically equivalent.

Reply:
Gotchas, we do checks for GroupEvent in the current [task](https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_apps/tasks/sentry_apps.py#L224) which was why I was curious if this was something to look into

Reply:
Right, yeah it's just because we were passing the actual `GroupEvent` before and wanted to handle it, but it's fine to just load it as an `Event` here too